src/hazelcore/Statics_Exp.re

```diff
@@ -45,29 +45,82 @@ and syn_block = (ctx: Contexts.t, block: UHExp.block): option(HTyp.t) => {
   let* ctx = syn_lines(ctx, leading);
   syn_opseq(ctx, conclusion);
 }
-and syn_lines =
-    (ctx: Contexts.t, lines: list(UHExp.line)): option(Contexts.t) => {
-  lines
-  |> List.fold_left(
-       (opt_ctx: option(Contexts.t), line: UHExp.line) => {
-         let* ctx = opt_ctx;
-         syn_line(ctx, line);
-       },
-       Some(ctx),
-     );
-}
-and syn_line = (ctx: Contexts.t, line: UHExp.line): option(Contexts.t) =>
-  switch (line) {
-  | EmptyLine
-  | CommentLine(_) => Some(ctx)
-  | ExpLine(opseq) =>
-    let+ _ = syn_opseq(ctx, opseq);
-    ctx;
-  | LetLine(p, def) =>
-    let def_ctx = extend_let_def_ctx(ctx, p, def);
-    let* ty_def = syn(def_ctx, def);
-    Statics_Pat.ana(ctx, p, ty_def);
-  }
+
+
+
+// e1 l1 a2 a3
+// [[e1]]
+// [[e1],[l1]]
+// [[e1],[l1,a2]]
+// [[e1],[l1,a2,a3]]
+
+// old accumulator: result
+// new accumulator: (crtGroup, groupList(determined))
+
+// removed .hd .tl
+
+and syn_lines = 
+    (ctx:Contexts.t, lines:list(UHExp.line)):option(Contexts.t) => {
+
+      // first pass: group lines
+
+      let groups = List.fold_left(
+        (accu: (list(UHExp.line),list(list(UHExp.line))), line: UHExp.line) => {
```

Comment:
instead of calling it `accu` you should name the two parts immediately using a pair pattern, `(cur_block, blocks)` or something like that

Reply:
modified